README.txt
Wx::Demo is a separate package for the wxPerl demo, previously bundled
with the wxPerl distribution.

Copyright (c) 2000-2007 Mattia Barbon.
This package is free software; you can redistribute it and/or
modify it under the same terms as Perl itself.

To install:

perl Makefile.PL
make
make test
make install

then run 'wxperl_demo.pl'; under Mac OS X you will need to run
'wxPerl -S wxperl_demo.pl'.

BUGS

The examples are sometimes inconsistent and incomplete. This will be fixed.
It is no worst than what was previously inside the wxPerl distribution.


Changes
Revision history for Perl application Wx::Demo.

0.23  TBR

0.22  Mon Mar 31 2014
    - compatibility with Wx 0.9923 and wxWidgets 3.0.0

0.21  Mon Apr 22 2013
    - Added wxCommandLinkButton demo
    - Fixes as per rt#84591

0.20  Wed Apr 03 2013
    - Added wxFontEnumerator demo
    - demostrate switching off Ubuntu scrollbar overlay

0.19  Tue Oct 02 2012
    - Updated for wxRibbonControl changes in Wx 0.9912
    - Demo for wxRearrangeCtrl and friends added

0.18  Fri Aug 24 2012
    - Added Wx::TreeEvent::GetKeyEvent example

0.17  Thu May 31 2012
    - Added Wx::PrintPaperDatabase

0.16  Tue Mar 27 2012
    - Prettier load fail for modules not supported by host Wx / wxWidgets 
    - Added wxWebView
    - Added wxTimePickerCtrl
    - Added wxRichToolTip
    - Added wxBannerWindow
    - Added wxInfoBar ( minimally )

0.15  Fri Mar 16 2012
    - Added wxTreeListCtrl
    - Added wxRibbonControl
    - Added wxPropertyGrid

0.14  Tue Feb 28 2012
    - Fix wxOverlay font on MSW + mingw + gdiplus
    - Add wxDF_UNICODETEXT handling in wxClipboard demo
    - several wxWidgets 2.9.3 compatibility fixes

0.13  Thu May 05 08:00:00 BST 2011
    - Wx::PlVlistBox example expanded
    - Add bitmap features of Wx::Button
    - Add Wx::GridTableBase - GetAttr

0.12  Wed Feb  2 13:30:00 CEST 2011    
    - don't use wxPerl on MacOSX if >= 0.98
    - Tests can pass if wxMediaCtrl is missing
    - Some Makefile.PL improvements (Gabor Szabo)
    - Added demo for wxHeaderCtrl and wxHeaderCtrlSimple.
    - Added demo for wxOverlay.

0.11  Fri Apr  2 18:34:04 CEST 2010
    - Allow searching inside the source code (Gabor Szabo).
    - Add another tab with the list of all wxPerl classes/events
      used in the demo (Gabor Szabo).
    - Added demo for wxSpinCtrlDouble.

0.10  Sun Jun 29 21:51:39 CEST 2008
    - Fixed wxListCtrl report mode demo latent bug.
    - Fixed running under Perl 5.10.

0.09  Sun Aug 26 01:10:56 CEST 2007
    - Added demo for wxHyperlinkCtrl.
    - Added demo for wxTipDialog.
    - Added demo for wxPasswordEntryDialog.

0.08  Thu Aug 23 21:56:10 CEST 2007
    - Added "New" category for demos new since last version.
    - Added demo for wxVScrolledWindow and wxHVScrolledWindow.
    - Added demo for wxComboCtrl.
    - Added demo for wxAboutDialog.
    - Added demo for wxCollapsiblePane.
    - Added demo for wxSearchControl.

0.07  Wed Mar 28 22:57:39 CEST 2007
    - Added demo for some of the wxWidgets help classes.

0.06  Thu Mar 22 00:08:44 CET 2007
    - Added demo for wxToolBar (moved the code from the wxPerl one).
    - Added demo for wxCaret (moved the code from the wxPerl one).
    - Added demo for wxArtProvider (moved the code from the wxPerl one).
    - Added demo for Document/View framework (contributed by Simon Flack
      in 2002).

0.05  Sun Nov 26 18:23:54 CET 2006
    - Added wxAuiNotebook to the AUI demo.
    - Added examples for more wxGrid editors/renderers.
    - Added demo for wxTreeCtrl.

0.04  Sun Nov 12 18:20:32 CET 2006
    - Added demo for Advanced User Interface (AUI).
    - Improved the wxRichTextCtrl demo.

0.03  Sat Nov 11 16:50:00 CET 2006
    - Added demos for the book and picker controls.
    - Added demo for wxRichTextCtrl.

0.02  Tue Aug 29 21:24:53 CEST 2006
    - Work with wxWidgets 2.5.3.

0.01  Sun Aug 27 17:30:34 CEST 2006
    - First public release.


lib/Wx/DemoHints/CoreHints.pm
#############################################################################
## Name:        lib/Wx/DemoHints/CoreHints.pm
## Purpose:     wxPerl demo hint helper for Wx::BannerWindow
## Author:      Mark Dootson
## Created:     26/03/2012
## RCS-ID:      $Id$
## Copyright:   (c) 2012 Mattia Barbon
## Licence:     This program is free software; you can redistribute it and/or
##              modify it under the same terms as Perl itself
#############################################################################

# this file contains hints for all the wx core modules
# for an example of a standalone file that an external
# module might add, see Wx/DemoHints/wxBannerWindow.pm

use strict;
use warnings;
use Wx;

package Wx::DemoHints::CoreHints;

our @hintpackages = ();

sub hint_packages { return @hintpackages; }

#------------------------------------------------------------

package 
    Wx::DemoHints::Base;

#------------------------------------------------------------

use strict;

use base qw(Wx::Panel);
use Wx qw(:sizer);

sub new {
    my( $class, $parent ) = @_;
    my $self = $class->SUPER::new( $parent );
    my $sizer = Wx::BoxSizer->new( wxVERTICAL );
    my $msg = $self->hint_message;
    my $display = Wx::StaticText->new($self, -1, $msg );
    $sizer->Add($display, 1, wxEXPAND|wxALL, 30);
    $self->SetSizer($sizer);
    return $self;
}

sub title { 'unknown' }

sub add_to_tags { qw( fail ) }

sub hint_message { 'this module could not be loaded'; }

sub file {
	my $rc = shift;
	my $class = ( ref($rc) ) ? ref($rc) : $rc;
	my $rfname = $class . '.pm';
	$rfname =~ s{::}{/}g;
	my $outname = $INC{$rfname};
	$outname =~ s{Wx/DemoHints/}{Wx/DemoModules/};
	return $outname;
}

#-------------------------------------------------------

package 
    Wx::DemoHints::CoreHintBase;

#-------------------------------------------------------
use base qw( Wx::DemoHints::Base );

sub register_hint { push(@Wx::DemoHints::CoreHints::hintpackages, $_[0]); }

sub file {
	my $rc = shift;
	my $class = ( ref($rc) ) ? ref($rc) : $rc;
	my $rfname = $class . '.pm';
	$rfname =~ s{::}{/}g;
	$rfname =~ s{Wx/DemoHints/}{Wx/DemoModules/};
	my $firstinc = $INC{'Wx/DemoHints/CoreHints.pm'};
	$firstinc =~ s{Wx/DemoHints/CoreHints.pm}{$rfname};
	return $firstinc if -e $firstinc;
	
	# maybe we are loaded from a different path to module
	# if we have a defautl rule for an external module
	my $checkpath;
	for my $incpath ( @INC ) {
		$checkpath = qq($incpath/$rfname);
		last if -e $checkpath;
	}
	unless( $checkpath && -e $checkpath ) {
		# we must return a filepath or Demo will crash
		# so return this file.
		# It shouldn't be possible to actually
		# get here - but .....
		$checkpath = $INC{'Wx/DemoHints/CoreHints.pm'};
	}
	return $checkpath;
}

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxPropertyGrid;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { eval { return Wx::_wx_optmod_propgrid(); }; }
sub title { 'wxPropertyGrid' }
sub hint_message { 'Wx::PropertyGrid requires Wx >= 0.9905 and wxWidgets >= 2.9.3'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxRichToolTip;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { eval { return Wx::_wx_optmod_propgrid(); }; }
sub title { 'wxRichToolTip' }
sub hint_message { 'Wx::RichToolTip requires Wx >= 0.9906 and wxWidgets >= 2.9.3'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxInfoBar;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { defined(&Wx::InfoBar::new); }
sub title { 'wxInfoBar' }
sub hint_message { 'Wx::InfoBar requires Wx >= 0.9906 and wxWidgets >= 2.9.3'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxTimePickerCtrl;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { defined(&Wx::TimePickerCtrl::new); }
sub title { 'wxTimePickerCtrl' }
sub hint_message { 'Wx::TimePicker requires Wx >= 0.9906 and wxWidgets >= 2.9.3'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxHeaderCtrl;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { defined &Wx::HeaderCtrl::new; }
sub title { 'wxHeaderCtrl' }
sub hint_message { 'Wx::PlHeaderCtrl requires wxWidgets >= 2.9.0'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxHeaderCtrlSimple;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { defined &Wx::HeaderCtrlSimple::new; }
sub title { 'wxHeaderCtrlSimple' }
sub hint_message { 'Wx::HeaderCtrlSimple requires wxWidgets >= 2.9.0'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxWebView;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { eval { return Wx::_wx_optmod_webview(); }; }
sub title { 'wxWebView' }
sub hint_message { 'Wx::WebView requires wxWidgets >= 2.9.3, Wx >= 0.9906 and an available backend on your system'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxRibbonControl;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { eval { return Wx::_wx_optmod_ribbon(); }; }
sub title { 'wxRibbonControl' }
sub hint_message { 'Wx::RibbonControl requires wxWidgets >= 2.9.3 and Wx >= 0.9905'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxTreeListCtrl;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { (Wx::wxVERSION < 2.009) }
sub title { 'wxTreeListCtrl' }
sub hint_message { 'Wx::TreeListCtrl from CPAN cannot work with wxWidgets >= 2.9.0'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxNativeTreeListCtrl;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { defined &Wx::PlTreeListItemComparator::new; }
sub title { 'wxTreeListCtrl (native)' }
sub hint_message { 'The Native Wx::TreeListCtrl requires Wx >= 0.9906 and wxWidgets >= 2.9.3. For wxWidgets 2.8.x you can use the Wx::TreeListCtrl module from CPAN.'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxGraphicsContext;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { defined &Wx::GraphicsContext::Create; }
sub title { 'wxGraphicsContext' }
sub hint_message { 'Your wxWidgets was not compiled with wxGraphicsContext support'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxOverlay;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { eval { my $olay = Wx::Overlay->new; }; ( $@ ) ? 0 : 1; }
sub title { 'wxOverlay' }
sub hint_message { 'Your Wx + wxWidgets version combination does not support wxOverlay'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxMediaCtrl;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { local $@; eval { require Wx::Media }; defined &Wx::MediaCtrl::new; }
sub title { 'wxMediaCtrl' }
sub hint_message { 'Your wxWidgets was not compiled with wxMediaCtrl support'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxPrintPaperDatabase;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { require Wx::Print; return defined(&Wx::PrintPaperDatabase::FindPaperTypeById); }
sub title { 'wxPrintPaperDatabase' }
sub hint_message { 'wxPrintPaperDatabase requires Wx >= 0.9909'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxRearrangeCtrl;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { return defined(&Wx::RearrangeCtrl::new); }
sub title { 'wxRearrangeCtrl' }
sub hint_message { 'wxRearrangeCtrl requires Wx >= 0.9914 and wxWidgets >= 2.9.4'; }

#---------------------------------------------------------------------------
package
	Wx::DemoHints::wxCommandLinkButton;
use base qw( Wx::DemoHints::CoreHintBase );
__PACKAGE__->register_hint;
sub can_load { return defined(&Wx::CommandLinkButton::new); }
sub title { 'wxCommandLinkButton' }
sub hint_message { 'wxCommandLinkButton requires Wx >= 0.9922 and wxWidgets >= 2.9.2'; }


1;

